codesign/test/codesign_test.dart

```diff
@@ -2,11 +2,64 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-import 'package:codesign/codesign.dart';
-import 'package:test/test.dart';
+import 'package:codesign/codesign.dart' as cs;
+import 'package:file/file.dart';
+import 'package:file/memory.dart';
+import './src/common.dart';
+
+/// A fake file visitor for testing purpose.
+///
+/// Upload for notarization is overriden, and the timer to check notarization
+/// status is fired instantly.
+class FakeCodesignVisitor extends cs.FileCodesignVisitor {
+  FakeCodesignVisitor({
+    required super.commitHash,
+    required super.codesignCertName,
+    required super.codesignUserName,
+    required super.appSpecificPassword,
+    required super.codesignAppstoreId,
+    required super.codesignTeamId,
+    required super.codesignFilepaths,
+    super.production,
+    this.fixitCheckMode = false,
+  });
+
+  bool fixitCheckMode;
+}
 
 void main() {
-  test('calculate', () {
-    expect(calculate(), 42);
+  const String randomString = 'abcd1234';
+  late MemoryFileSystem fileSystem;
+  fileSystem = MemoryFileSystem.test();
+  TestStdio stdio = TestStdio();
+  List<String> fakeFilepaths = ['a.zip', 'b.zip', 'c.zip'];
```

Comment:
This can be const
```suggestion
  const List<String> fakeFilepaths = ['a.zip', 'b.zip', 'c.zip'];
```

Reply:
https://dart-lang.github.io/linter/lints/prefer_const_constructors.html

Reply:
True, added lint packages.